CONTRIBUTING.md

```diff
@@ -0,0 +1,198 @@
+-   [pytest](https://docs.pytest.org/en/)
+
+The development version of Python is 3.12, so please make sure to use that version when developing.
+The client-side code is tested against Python 3.10 and 3.11, server-side code is tested against Python 3.11, please take this into account when using recent Python features.
+
+To get started with development, [create a fork](https://github.com/aai-institute/jobq/fork) of the GitHub repository and clone it to your local machine.
+Please submit your changes as pull requests against the `main` branch.
+
+## Working on the client-side code (decorators & CLI)
+
+The `client/` directory contains the source code for the `jobq` CLI and Python decorators.
+
+### Development
+
+If you want to contribute to the client-side code, you can follow these steps:
+
+1. Create a virtual environment and install the development dependencies:
+
+    ```shell
+    cd jobq/client
+    uv venv
+    source .venv/bin/activate
+    uv pip install -r requirements-dev.txt
+    uv pip install -e . --no-deps
+    ```
+
+2. To run the Pytest test suite, run:
+
+    ```shell
+    pytest
+    ```
+
+3. After making your changes, verify they adhere to our Python code style by running `pre-commit`:
+
+    ```shell
+    pre-commit run --all-files
+    ```
+
+    You can also set up Git hooks through `pre-commit` to perform these checks automatically:
+
+    ```shell
+    pre-commit install
+    ```
+
+### Regenerating the API client
+
+The `src/openapi_client` folder contains an automatically-generated API client for the backend API.
+
+If you make changes to the backend API, you can regenerate the API client with the following command:
+
+```console
+hack/openapi-regen.sh
+```
+
+This will regenerate the API client in the `client/src/openapi_client` directory from a currently running FastAPI server using [`openapi-generator-cli`](https://openapi-generator.tech/).
+Note that you will need to have the backend server running and accessible at `http://localhost:8000` in order to generate the client code.
+
+The script automatically removes unnecessary files and reformats the generated code according to our code style.
+
+### Publishing to PyPI
+
+The `jobq` package is published to [PyPI](https://pypi.org/project/jobq/) through a GitHub Actions workflow when a new release is created.
+
+## Working on the server-side code (API)
+
+The server-side code under the `backend/` folder is written in Python and uses the [FastAPI](https://fastapi.tiangolo.com/) framework.
+
+You can follow the same instructions as for the client-side code to set up a development environment.
+
+### Running the server
+
+Since the code can load Kubernetes credentials from an in-cluster Kubernetes service account or from a Kubeconfig file, you can run it locally without having to deploy to a Kubernetes cluster.
+
+To run the server locally in development mode (accessible at <http://localhost:8000>), you can use the following command in the `backend/` folder:
```

Comment:
Is this rendered correctly in the docs, or does some HTML interpreter screw this up?

Reply:
It renders as a simple link, seems like this is beyond the highlighter's capabilities.